cephprimarystorage/test_watcher_parse.py

```
@@ -0,0 +1,175 @@
+        raw, infos = _parse_watchers("pool/vol-001", shell_fn)
+
+        self.assertEqual(len(raw), 0)
+        self.assertEqual(len(infos), 0)
+
+    def test_text_fallback_single_watcher(self):
+        """When JSON fails, fall back to text parsing."""
+        text_output = "Watchers:\n\twatcher=10.0.0.5:0/999 client.123 cookie=1\n"
+
+        def shell_fn(cmd):
+            if "--format json" in cmd:
+                raise Exception("not supported")
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)